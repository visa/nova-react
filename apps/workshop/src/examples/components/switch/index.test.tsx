/**
 *              Â© 2025 Visa
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *         http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 **/
import { render } from '@testing-library/react';
import { axe } from 'jest-axe';

import metaData from './meta.json';

import { DefaultSwitch } from './default-switch';
import { DisabledSwitch } from './disabled-switch';
import { OptionalMessageSwitch } from './optional-message-switch';
import { DisabledSwitchOn } from './disabled-switch-on';

const examples = [
  { Component: DefaultSwitch, title: metaData['default-switch'].title },
  { Component: DisabledSwitch, title: metaData['disabled-switch'].title },
  { Component: DisabledSwitchOn, title: metaData['disabled-switch-on'].title },
  { Component: OptionalMessageSwitch, title: metaData['optional-message-switch'].title },
];

describe('Switch examples', () => {
  examples.forEach(({ Component, title }) => {
    it(`${title} should render correctly`, async () => {
      const { container } = render(<Component />);
      const results = await axe(container);
      expect(results).toHaveNoViolations();
      expect(container).toMatchSnapshot();
    });
  });
});
