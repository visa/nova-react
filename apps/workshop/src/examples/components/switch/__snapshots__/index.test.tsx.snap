// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch examples Default switch should render correctly 1`] = `
<div>
  <div
    class="v-flex v-flex-wrap v-gap-10 v-justify-content-between v-m-8"
    style="max-inline-size: 288px;"
  >
    <label
      class="v-label v-switch-label"
      for="default-switch-example-switch"
    >
      Label
    </label>
    <input
      class="v-switch"
      id="default-switch-example-switch"
      name="default-switch"
      role="switch"
      type="checkbox"
    />
  </div>
</div>
`;

exports[`Switch examples Disabled selected switch should render correctly 1`] = `
<div>
  <div
    class="v-flex v-flex-wrap v-gap-10 v-justify-content-between v-m-8"
    style="max-inline-size: 288px;"
  >
    <label
      class="v-label v-switch-label"
      for="disabled-switch-on-example-switch"
    >
      Label
    </label>
    <input
      checked=""
      class="v-switch"
      disabled=""
      id="disabled-switch-on-example-switch"
      name="disabled-switch-on"
      role="switch"
      type="checkbox"
    />
  </div>
</div>
`;

exports[`Switch examples Disabled switch should render correctly 1`] = `
<div>
  <div
    class="v-flex v-flex-wrap v-gap-10 v-justify-content-between v-m-8"
    style="max-inline-size: 288px;"
  >
    <label
      class="v-label v-switch-label"
      for="disabled-switch-example-switch"
    >
      Label
    </label>
    <input
      class="v-switch"
      disabled=""
      id="disabled-switch-example-switch"
      name="disabled-off-switch"
      role="switch"
      type="checkbox"
    />
  </div>
</div>
`;

exports[`Switch examples Switch with description should render correctly 1`] = `
<div>
  <div
    class="v-align-items-center v-flex v-flex-row v-flex-wrap v-gap-10 v-justify-content-between v-m-8"
    style="max-inline-size: 288px;"
  >
    <div
      class="v-flex v-flex-col v-flex-grow v-gap-2 v-justify-content-between"
    >
      <label
        class="v-label v-switch-label"
        for="optional-message-switch-example-switch"
      >
        Label
      </label>
      <span
        class="v-input-message"
        id="optional-message-switch-example-message"
      >
        This is optional text that can be used to describe the label in more detail.
      </span>
    </div>
    <input
      aria-describedby="optional-message-switch-example-message"
      class="v-switch"
      id="optional-message-switch-example-switch"
      name="switch-with-message"
      role="switch"
      type="checkbox"
    />
  </div>
</div>
`;
